### PR TITLE
Set certain filter keys as hardcoded and streamline filtering flow

### PR DESCRIPTION
**Pull request by mayan-000**
## Description
This pull request updates certain keys which were calculated from cookie data to hard coded ones(SameSite, HttpOnly, Cookie Accepted), streamlines the filtering mechanism's flow between files and adds enum for retention period filters.

## Screenshot/Screencast
Screenshot 2023-08-23 at 10 41 10 AM

---

Fixes #

### CODE REVIEW

File: packages/extension/src/view/devtools/stateProviders/filterManagementStore/constants.ts [hunk 1]
```diff
@@ -56,33 +39,53 @@ export const FILTER_MAPPING = [
...
-    name: 'Cookie Accepted',
-    keys: 'isCookieSet',
-    type: 'boolean',
-    order: 10,
-    description:
-      "Whether the cookie was accepted(set) in Chrome's Cookie Store",
-  },
 ];
 
+export enum RetentionPeriodFilter {
```
Comment (sayedtaqui) [inline packages/extension/src/view/devtools/stateProviders/filterManagementStore/constants.ts:44]: JFI, we are planning to add Expired as well.

File: packages/extension/src/view/devtools/stateProviders/filterManagementStore/utils/filterCookiesWithCustomMapping.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+ * @param value Value of the cookie key to be filtered
+ * @returns boolean whether the cookie should be shown or not
+ */
+function filterWithScope(selectedFilter: Set<string>, value: string) {
+  let canShow = false;
+  if (selectedFilter.has('Third Party') && !selectedFilter.has('First Party')) {
+    canShow = !value;
+  } else if (
+    !selectedFilter.has('Third Party') &&
+    selectedFilter.has('First Party')
```
Comment (sayedtaqui) [inline packages/extension/src/view/devtools/stateProviders/filterManagementStore/utils/filterCookiesWithCustomMapping.ts:64]: Not to be done immediately, but I think we shouldn't use text strings in logic for any filter labels like we were doing for retention period.

**Review by sayedtaqui — COMMENTED** [review]
@mayan-000 Is the QA done for this PR?

File: packages/extension/src/view/devtools/components/tabContent/cookies/cookieFilter/chips/chip.tsx [hunk 3]
```diff
@@ -25,7 +25,10 @@ import type {
   SelectedFilters,
   Filter,
 } from '../../../../../stateProviders/filterManagementStore/types';
-import { FILTER_MAPPING } from '../../../../../stateProviders/filterManagementStore/constants';
+import {
+  CUSTOM_FILTER_MAPPING,
+  FILTER_MAPPING,
+} from '../../../../../stateProviders/filterManagementStore/constants';
```
Comment (ayushnirwal) [inline packages/extension/src/view/devtools/components/tabContent/cookies/cookieFilter/chips/chip.tsx:31]: Could you please add types and constants to the index file of filter management store so these can be imported with one statement

File: packages/extension/src/view/devtools/stateProviders/filterManagementStore/utils/filterCookiesWithCustomMapping.ts [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+ * @param selectedFilter Set<string> of selected filters
+ * @param value Value of the cookie key to be filtered
+ * @returns boolean whether the cookie should be shown or not
+ */
+function filterWithScope(selectedFilter: Set<string>, value: string) {
+  let canShow = false;
+  if (selectedFilter.has('Third Party') && !selectedFilter.has('First Party')) {
+    canShow = !value;
+  } else if (
+    !selectedFilter.has('Third Party') &&
```
Comment (ayushnirwal) [inline packages/extension/src/view/devtools/stateProviders/filterManagementStore/utils/filterCookiesWithCustomMapping.ts:63]: I think using enums would be beneficial similar to retention period filtering.

File: packages/extension/src/view/devtools/stateProviders/filterManagementStore/utils/matchTerm.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  filteredCookies: { [key: string]: CookieTableData }
+) {
+  const cookieName = cookieData.parsedCookie.name;
+
+  if (!searchTerm) {
+    filteredCookies[cookieName] = cookieData;
+  }
+
+  const lowerCaseTerm = searchTerm.toLowerCase();
+  const didSearchTermMatch =
```
Comment (ayushnirwal) [inline packages/extension/src/view/devtools/stateProviders/filterManagementStore/utils/matchTerm.ts:39]: @sayedtaqui Did we plan to only search through name and domain?

File: packages/extension/src/view/devtools/stateProviders/filterManagementStore/constants.ts [hunk 6]
```diff
@@ -56,33 +39,53 @@ export const FILTER_MAPPING = [
...
+  httpOnly: {
+    name: 'HttpOnly',
+    keys: 'parsedCookie.httponly',
+    filters: new Set(['True', 'False']),
+    order: 4,
+  },
+  secure: {
+    name: 'Secure',
+    keys: 'parsedCookie.secure',
+    filters: new Set(['True', 'False']),
```
Comment (ayushnirwal) [inline packages/extension/src/view/devtools/stateProviders/filterManagementStore/constants.ts:74]: Do these need to be a set now if they are hard coded?

**Review by ayushnirwal — CHANGES_REQUESTED** [review]

File: packages/extension/src/view/devtools/stateProviders/filterManagementStore/utils/matchTerm.ts (hunk 5 above)
Comment (sayedtaqui) [reply packages/extension/src/view/devtools/stateProviders/filterManagementStore/utils/matchTerm.ts:39]: As of now yes, any suggestions?

**Comment by mayan-000** [comment]
I will reopen this PR with further changes as per bandwidth availability.

**Pull request closed without merging**